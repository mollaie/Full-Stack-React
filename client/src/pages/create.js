import React from "react";
import { useNavigate } from "react-router-dom";
import { Formik, Field } from "formik";
import * as Yup from "yup";
import Layout from "../components/layout";

const CreateContainer = () => {
  let navigate = useNavigate();

  return (
    <div className="main-container">
      <Formik
        initialValues={{ name: "", store: "" }}
        onSubmit={(values, { setSubmitting }) => {
          setTimeout(() => {
            setSubmitting(false);
            navigate("/Dashboard");
          }, 500);
        }}
        validationSchema={Yup.object().shape({
          username: Yup.string().required("Required"),
          password: Yup.string()
            .required("Required")
            .min(6, "Password is too short - should be 6 characters minimum"),
        })}
      >
        {(props) => {
          const {
            values,
            touched,
            errors,
            isSubmitting,
            handleChange,
            handleBlur,
            handleSubmit,
          } = props;

          return (
            <form onSubmit={handleSubmit}>
              <div className="item-container">
                <label htmlFor="name">Reservation name</label>
                <input
                  name="name"
                  type="text"
                  placeholder="Please enter your reservation name"
                  value={values.name}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  className={errors.name && touched.name && "error"}
                />
                {errors.name && touched.name && (
                  <div className="input-feedback">{errors.name}</div>
                )}
              </div>

              <div className="item-container">
                <label htmlFor="store">Store</label>
                <select
                  name="store"
                  value={values.store}
                  onChange={handleChange}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  className={errors.username && touched.username && "error"}
                >
                  <option>Store A</option>
                  <option>Store B</option>
                  <option>Store C</option>
                </select>
                {errors.store && touched.store && (
                  <div className="input-feedback">{errors.store}</div>
                )}
              </div>

              <div className="item-container">
                <label htmlFor="status">Status</label>
                <div className="radio-container" role="group">
                  <label>
                    <Field type="radio" name="picked" value="Todo" />
                    Todo
                  </label>
                  <label>
                    <Field type="radio" name="picked" value="In Progress" />
                    In Progress
                  </label>

                  <label>
                    <Field type="radio" name="picked" value="Ready" />
                    Ready
                  </label>
                </div>
                {errors.status && touched.status && (
                  <div className="input-feedback">{errors.status}</div>
                )}
              </div>
            </form>
          );
        }}
      </Formik>
    </div>
  );
};

const Create = () => {
  let props = {
    isCreate: true,
  };
  return (
    <Layout isCreate={true}>
      <CreateContainer />
    </Layout>
  );
};

export default Create;
