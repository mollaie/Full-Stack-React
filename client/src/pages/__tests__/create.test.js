import React from "react";
import renderer from "react-test-renderer";
import create from "../create";

describe("create test", () => {
  it("create should match snapshot", () => {
    const component = renderer.create(<create />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
