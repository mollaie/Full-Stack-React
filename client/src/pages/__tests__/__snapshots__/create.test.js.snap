// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create test create should match snapshot 1`] = `<create />`;
